packages/vite/src/node/optimizer/index.ts

```diff
@@ -496,22 +498,153 @@ export async function runOptimizeDeps(
-      fs.rmSync(processingCacheDir, { recursive: true, force: true })
+    cancel: cleanUp,
+  })
+
+  if (!qualifiedIds.length) {
+    return {
+      cancel: async () => cleanUp(),
+      result: Promise.resolve(createProcessingResult()),
+    }
+  }
+
+  const start = performance.now()
+
+  const preparedRun = prepareEsbuildOptimizerRun(
+    resolvedConfig,
+    depsInfo,
+    ssr,
+    processingCacheDir,
+  )
+
+  const result = preparedRun.then(({ context, idToExports }) => {
+    return context
+      .rebuild()
+      .then((result) => {
+        const meta = result.metafile!
+
+        // the paths in `meta.outputs` are relative to `process.cwd()`
+        const processingCacheDirOutputPath = path.relative(
+          process.cwd(),
+          processingCacheDir,
+        )
+
+        for (const id in depsInfo) {
+          const output = esbuildOutputFromId(
+            meta.outputs,
+            id,
+            processingCacheDir,
+          )
+
+          const { exportsData, ...info } = depsInfo[id]
+          addOptimizedDepInfo(metadata, 'optimized', {
+            ...info,
+            // We only need to hash the output.imports in to check for stability, but adding the hash
+            // and file path gives us a unique hash that may be useful for other things in the future
+            fileHash: getHash(
+              metadata.hash +
+                depsInfo[id].file +
+                JSON.stringify(output.imports),
+            ),
+            browserHash: metadata.browserHash,
+            // After bundling we have more information and can warn the user about legacy packages
+            // that require manual configuration
+            needsInterop: needsInterop(
+              config,
+              ssr,
+              id,
+              idToExports[id],
+              output,
+            ),
+          })
+        }
+
+        for (const o of Object.keys(meta.outputs)) {
+          if (!o.match(jsMapExtensionRE)) {
+            const id = path
+              .relative(processingCacheDirOutputPath, o)
+              .replace(jsExtensionRE, '')
+            const file = getOptimizedDepPath(id, resolvedConfig, ssr)
+            if (
+              !findOptimizedDepInfoInRecord(
+                metadata.optimized,
+                (depInfo) => depInfo.file === file,
+              )
+            ) {
+              addOptimizedDepInfo(metadata, 'chunks', {
+                id,
+                file,
+                needsInterop: false,
+                browserHash: metadata.browserHash,
+              })
+            }
+          }
+        }
+
+        const dataPath = path.join(processingCacheDir, '_metadata.json')
+        writeFile(
+          dataPath,
+          stringifyDepsOptimizerMetadata(metadata, depsCacheDir),
+        )
+
+        debug(`deps bundled in ${(performance.now() - start).toFixed(2)}ms`)
+
+        return createProcessingResult()
+      })
+      .finally(() => {
+        context.dispose()
```

Comment:
```suggestion
        return context.dispose()
```
`context.dispose` seems to return a promise.

Reply:
returning another Promise from finally can hide the original rejection if that is rejected, see 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally  

> If the handler throws an error or returns a rejected promise, the promise returned by finally() will be rejected with that value instead. Otherwise, the return value of the handler does not affect the state of the original promise

Reply:
Oh, I didn't know about that. Maybe like this is better?
```js
return context.dispose().catch(e => { console.error('error happed during context.dispose', e) })
```

Reply:
Applied the same to the scan process here [`1b700b5` (#12237)](https://github.com/vitejs/vite/pull/12237/commits/1b700b5613d25337037c6da6b03808644df1ed59)